config/routes.rb

```diff
@@ -2,5 +2,10 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  # root "users#index"
+  get 'users/:user_id/posts', to: 'posts#index', as: :users_posts
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: :user_post
+  get 'users', to: 'users#index', as: :users
+  get 'users/:id', to: 'users#show', as: :user
+
```

Comment:
- [ ] Instead of manually typing out all these routes we can optimize these routes definitions using rails helper `resources` and the `only` keyword to limit exposed routes. See example

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```

Reply:
@DammyShittu Thank you for your feedback. I will implement feedback.